internal/cluster/clustersecret/store.go

```diff
@@ -0,0 +1,86 @@
+// Copyright © 2019 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clustersecret
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+)
+
+// NewSecret represents a new secret.
+type NewSecret struct {
```

Comment:
i don't like this terminology. this is not a secret, this is only the input for creating one. and the term also collides with all the New* functions we have here. request is easier to understand

Reply:
Well, it **is** a secret, a new one which is not persisted yet. I removed request on purpose, because this is not a request-response scenario, but I see how it might be confusing, so I will try to come up with something better. Feel free to suggest alternatives though.